### PR TITLE
core-datepicker: disable month only if all days are disabled

### LINKED ISSUE #386: core-datepicker: Entire month disabled if month contains a disabled date with the same day of the month as the selected date
Testcase: https://codesandbox.io/s/blue-pond-nz8z1

Relevant code from the testcase:
```js
const datePicker = document.querySelector("core-datepicker");

// Select a date in may 2019. Note that the day of the month is `06`
const selectedDate = new Date("2019-05-06");
datePicker.date = selectedDate;

// Disable a date in september 2019.
// Note that the day of the month is `06`, same as the selected date.
// `2019-09-07` for example does not cause the month to be disabled
// from the date dropdown.
const disabledDate = new Date("2019-09-06");
datePicker.disabled = date => {
  return date.valueOf() === disabledDate.valueOf();
};
```

### PR DESCRIPTION
**Pull request by laat**
fixes #386

### CODE REVIEW

File: packages/core-datepicker/core-datepicker.js [hunk 1]
```diff
@@ -155,7 +155,22 @@ function select (self, select) {
...
+  const month = date.getMonth()
+  const days = []
+  while (date.getMonth() === month) {
+    days.push(new Date(date))
+    date.setDate(date.getDate() + 1)
+  }
+  return days
+}
+function isMonthDisabled (dateInMonth, isDayDisabled) {
+  return daysInMonth(dateInMonth).map(isDayDisabled).reduce((a, b) => a && b)
```
Comment (esseb) [inline packages/core-datepicker/core-datepicker.js]: Could use `.every(day)`, but I don't know if there is a polyfill reason not to use it.

Comment (laat) [reply packages/core-datepicker/core-datepicker.js]: I did not want to use any polyfill. The polyfill is huge, and map.reduce over arrays of 31 days does not need that optimization

Comment (esseb) [reply packages/core-datepicker/core-datepicker.js]: Right, reduce is better supported than I thought.

**Comment by laat** [comment]
I see. This patch did not take into account that the select can select anything, dates months years millennia.

This means that we need to extend how we model options.

---
maybe granularity on option?
```html
<option value="${new Date()}" granularity="month|day|year"
```

---
maybe granularity on select?
```html
<select granularity="month|day|year">
```
---
Or even more generic by adding a function to select?
```html
<select>...</select>
```
```js
document.querySelector('select').optionDisabled = date => date.getMonth !== 9
```

@eirikbacker thoughts?

**Comment by esseb** [comment]
@laat my use case could be solved by a callback when selecting a month option where I receive the date the date picker wants to select as a parameter and I have the option to return a different date I want to select instead.

**Pull request closed without merging**